impl/maven-impl/src/main/java/org/apache/maven/impl/DefaultModelXmlFactory.java

```diff
@@ -121,7 +121,7 @@ public void write(XmlWriterRequest<Model> request) throws XmlWriterException {
         Writer writer = request.getWriter();
         Function<Object, String> inputLocationFormatter = request.getInputLocationFormatter();
         if (writer == null && outputStream == null && path == null) {
-            throw new IllegalArgumentException("writer, outputStream or path must be non null");
```

Comment:
this is when being consequent considered smell, as throwing IAE but having NPE.

https://github.com/apache/maven/pull/2277#issuecomment-2833670455